lib/rules/eqeqeq.js

```diff
@@ -98,8 +98,9 @@ module.exports = {
          * @private
          */
         function isNullCheck(node) {
-            return (node.right.type === "Literal" && node.right.value === null) ||
```

Comment:
Why is this change needed?
I fear that this bug fix will create issues of people based on this line